Prox/Prox/PlaceCarousel/PlaceCarousel.swift

```diff
@@ -70,17 +72,49 @@ extension PlaceCarousel: UICollectionViewDataSource {
         cell.tripAdvisorReview.numberOfReviewersLabel.text = "6,665 Reviews"
         cell.tripAdvisorReview.reviewSiteLogo.image = UIImage(named: "logo_ta")
 
-        if indexPath.item == 0 {
-            cell.locationImage.image = UIImage(named: "icon_location")?.withRenderingMode(UIImageRenderingMode.alwaysTemplate)
-            cell.location.text = "You're here"
-            cell.isSelected = true
-        } else {
-            cell.location.text = "17 min walk away"
-            cell.isSelected = false
+        if let travelTimes = place.travelTimes {
+            self.setTravelTimes(travelTimes: travelTimes, forCell: cell)
+        } else if let currentLocation = self.currentLocation {
+            TravelTimesProvider.travelTime(fromLocation: currentLocation.coordinate, toLocation: place.latLong) { travelTimes in
+                place.travelTimes = travelTimes
+
+                DispatchQueue.main.async {
+                    self.setTravelTimes(travelTimes: travelTimes, forCell: cell)
+                }
+            }
         }
         return cell
     }
 
+    private func setTravelTimes(travelTimes: TravelTimes?, forCell cell: PlaceCarouselCollectionViewCell) {
+        guard let travelTimes = travelTimes else {
+            return
+        }
+
+        if let walkingTimeSeconds = travelTimes.walkingTime {
+            let walkingTimeMinutes = Int(round(walkingTimeSeconds / 60.0))
+            if walkingTimeMinutes <= 30 {
+                if walkingTimeMinutes < 3 {
+                    cell.locationImage.image = UIImage(named: "icon_location")?.withRenderingMode(UIImageRenderingMode.alwaysTemplate)
+                    cell.location.text = "You're here"
+                    cell.isSelected = true
```

Comment:
Since we may get more than one location that we're at, I'm not sure we want to use that as the selection criteria.